src/items/crud/homepage.py

```diff
@@ -511,7 +511,7 @@ async def update_homepage_elements(
 
                     component_elements_full = copy.deepcopy(details.component_elements) if copy.deepcopy(details.component_elements) else []
 
-                    component_detail = await self.get_component_element_details(db=db, component_elements_type=component_elements_type, component_elements=component_elements)
+                    component_detail = await self.get_component_element_details(db=db, component_elements_type=component_elements_type, component_elements=component_elements, filter_details=filter_details)
```

Comment:
why this ?

Reply:
filter_details was not being passed for update_type="insert_element" while it was being passed for add_element, replace_element etc. Incase, we insert_element of element_type filter, it would always return None and operation would fail